src/sentry/testutils/hybrid_cloud.py

```diff
@@ -196,12 +202,9 @@ def new_atomic_on_commit(func, using=None):
         _old_transaction_on_commit(func, using)
         maybe_flush_commit_hooks(transaction.get_connection(using))
 
-    for db_name in settings.DATABASES:
-        simulated_transaction_watermarks.state[
-            db_name
-        ] = simulated_transaction_watermarks.get_transaction_depth(
-            transaction.get_connection(db_name)
-        )
+    if is_django_test_case:
+        for db_name in settings.DATABASES:
+            simulated_transaction_watermarks.state[db_name] = 2
```

Comment:
Why is the starting depth 2 to start?

Reply:
Check the updated commit -- I added a comment that explains this, and some more tests!